services/provider/server/server.go

```diff
@@ -206,3 +212,121 @@ func Start(port int, providerServer *OCSProviderServer, opts []grpc.ServerOption
 		klog.Fatalf("failed to start gRPC server: %v", err)
 	}
 }
+
+func (s *OCSProviderServer) getJSONConnectionDetails(consumerResource *ocsv1alpha1.StorageConsumer) ([]externalSC.ExternalResource, error) {
+	var j []externalSC.ExternalResource
+
+	// Configmap with mon endpoints
+	configmap, err := s.consumerAPI.GetConfigMap("rook-ceph-mon-endpoints", s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-mon-endpoints` configMap. Error %v", err)
+	}
+
+	cmData := strings.Split(configmap.Data["data"], ",")
```

Comment:
So that's how you get the first mon? Perhaps adding a comment will help to understand what's going on here.
Also add a new var for the mon address so it's clearer.

Reply:
added.